Develop

arbulu89 opened the pull request:
PR to merge to master to create a new `v6` release. Main changes:

- Restructure salt and pillar folders usage, to have an easier usage
- Improve ISCSI module usage, making it optional
- Add support for zip,rar,exe,sar HANA media files
- Add new option to choose between HA or non HA deployments
- Implement bastion host usage for Azure
- Change HANA disks configuration usage for Azure, providing more flexibility and better default options
- Add hana active/active scenario usage
- Add the option to select the logging level and print salt result in terraform output

Variables usage update:

Removed:
- devel_mode
- init_type
- hana_disk_device
- iscsidev
- min_instancetype (aws only)
- hana_data_disk_type (azure only)
- hana_data_disk_size (azure only)
- hana_data_disk_caching (azure only)
- shared_storage_type (libvirt onlye)
- drbd_count (libvirt only)
- drbd_shared_storage_type (libvirt only)

Added:
- public_key_location (libvirt only)
- sbd_storage_type (libvirt only)
- bastion_enabled (azure only)
- bastion_public_key_location (azure only)
- bastion_private_key_location (azure only)
- hana_data_disk_configuration (azure_only)
- netweaver_nfs_share (azure only)
- provisioning_log_level
- hana_cluster_sbd_enabled
- hana_ha_enabled
- hana_active_active
- hana_cluster_vip_secondary
- drbd_cluster_sbd_enabled
- netweaver_cluster_sbd_enabled
- netweaver_inst_folder
- netweaver_ha_enabled
- iscsi_disk_size

Updated:
- instancetype -> hana_instancetype
- hdbserver_sar -> hana_archive_file
- iscsi_disk -> iscsi_lun_count
- netwaver_swpm_extract_dir -> netweaver_extract_dir
- netweaver_vm_size -> netweaver_xscs_vm_size and netweaver_app_vm_size (azure only)
- netweaver_enable_accelerated_networking -> netweaver_xscs_accelerated_networking and netweaver_app_accelerated_networking (azure only)

MalloZup reviewed (approved):
good work @arbulu89  and all who contributed to V6 :champagne: :champagne: :champagne: 


My concern are following now are the release note:

- we should clearly state which var are `deprecated`, which one are a causing any breaking change.

A part of this, I do think we are good since we have being using this a while

arbulu89 commented:
> good work @arbulu89 and all who contributed to V6 champagne champagne champagne
> 
> My concern are following now are the release note:
> 
>     * we should clearly state which var are `deprecated`, which one are a causing any breaking change.
> 
> 
> A part of this, I do think we are good since we have being using this a while

I will do changes diff of the variables file to put the new, updated and removed variables. I will publish this in the release notes

Simranpal reviewed (approved):
Thanks for the changes, looks good!
